### DISCUSSION TITLE
GeoDMS axis order, and projection based domain check

### DISCUSSION THREAD

**Discussion by eoudejans** [ObjectVision/GeoDMS#136] [Ideas]
Observation that the GeoAcademy defines col row (xy) in config.ini whereas default GeoDMS uses row col order (yx).
col row (xy) order seems to be more natural to most users and is how most projections are defined (with exception of for instance EPGS:4326, which defines north, east: lat, lon: yx order)
We could choose to follow gdal that always follows the authority (EPGS, ESRI, PROJ etc) projection definition.

**Comment by eoudejans** [comment]
Furthermore, could the GeoDMS domain checks be extended by also incorporating the authority projection definition bounding box?

**Comment by eoudejans** [comment]
Note that the divergence between GeoAcademy axis order and the default GeoDMS axis order may lead to major confusion when combining model code from the GeoAcademy and code from another model that uses the default GeoDMS order.

**Comment by MaartenHilferink** [comment]
Also, a client used a raster->vector example from a wiki, resulting in a 1 diagonal-raster size translation and parity flip due to not having the assumed [general] ConfigPointColRow=1 in the config.ini of the configuration's related sub-folder.
![image](https://user-images.githubusercontent.com/2284361/221208859-68913be2-c2c3-4e3b-b874-be960aa4fa58.png)

**Comment by MaartenHilferink** [comment]
new syntax: projection(coordinate, coordinate), f.e.: 
```
rdc_meter(150000f, 450000f); 
latlong(53.0, 3.0)
```
Or, additionally: geopoint(coordinatre, coordinate, rdc_meter)

Direct value definitions, such as: 

`attribute<rdc_meter> locations(city): [ (150000, 450000), (100000, 400000) ];`

should be replaced by 

`attribute<rdc_meter> locations(city) := [city: rdc_meter(150000, 450000), rdc_meter(100000, 400000) ];`

which is syntactic sugar for:

`attribute<rdc_meter> locations(city) := union_data(city, rdc_meter(150000, 450000), rdc_meter(100000, 400000));`

**Comment by MaartenHilferink** [comment]
also, consider 
`attribute<rdc_meter> locations(city) := [(150000, 450000), (100000, 400000) ];`
where [...] as main expr is replaced by a new-data-array call with domain and values unit derived from the context whereas the old syntax (without '=') will generate a 'warning: depreciated syntax depends on the default order of coordinates, consider using := to order coordinates according to the specification of the used projection".

**Comment by jipclaassens** [comment]
gerelateerd #46
